src/Sarif.Converters/FortifyFprConverter.cs

```diff
@@ -0,0 +1,631 @@
+
+                    var acl = new AnnotatedCodeLocation
+                    {
+                        Step = step++,
+                        PhysicalLocation = physLoc
+                    };
+
+                    // Remember the id of the snippet associated with this location.
+                    // We'll use it to fill the snippet text when we read the Snippets element later on.
+                    if (!string.IsNullOrEmpty(snippetId))
+                    {
+                        _aclToSnippetIdDictionary.Add(acl, snippetId);
+                    }
+
+                    codeFlow.Locations.Add(acl);
+
+                    // Keep track of the snippet associated with the last location in the
+                    // CodeFlow; that's the snippet that we'll associate with the Result
+                    // as a whole.
+                    lastSnippetId = snippetId;
+
+                    // Step past the empty element.
+                    _reader.Read();
+                }
+                else
+                {
+                    _reader.Read();
+                }
+            }
+
+            if (codeFlow.Locations.Any())
+            {
+                result.Locations.Add(new Location
+                {
+                    // TODO: Confirm that the traces are ordered chronologically
+                    // (so that we really do want to use the last one as the
+                    // overall result location).
+                    ResultFile = codeFlow.Locations.Last().PhysicalLocation
+                });
+
+                if (!string.IsNullOrEmpty(lastSnippetId))
+                {
+                    _resultToSnippetIdDictionary.Add(result, lastSnippetId);
+                }
+            }
+        }
+
+        private PhysicalLocation ParsePhysicalLocationFromSourceInfo()
+        {
+            string path = _reader.GetAttribute(_strings.PathAttribute);
+
+            int startLine;
+            string lineAttr = _reader.GetAttribute(_strings.LineAttribute);
+            if (!int.TryParse(lineAttr, out startLine))
+            {
+                startLine = 0;
+            }
+
+            int startColumn;
+            string colStartAttr = _reader.GetAttribute(_strings.ColStartAttribute);
+            if (!int.TryParse(colStartAttr, out startColumn))
+            {
+                startColumn = 0;
+            }
+
+            int endColumn;
+            string colEndAttr = _reader.GetAttribute(_strings.ColEndAttribute);
+            if (!int.TryParse(colEndAttr, out endColumn))
+            {
+                endColumn = 0;
+            }
+
+            return new PhysicalLocation
+            {
+                Uri = new Uri(path, UriKind.Relative),
+                Region = new Region
+                {
+                    StartLine = startLine,
+                    StartColumn = startColumn,
+                    EndColumn = endColumn
```

Comment:
Yes, in the only samples I've looked at, the SourceLine element has only "line", "colStart", and "colEnd" attributes. From the fact that the attribute is _not_ named "lineStart", I suspect that we won't find any samples with a "lineEnd" attribute.

---
In reply to: [94535925](https://github.com/Microsoft/sarif-sdk/pull/554#discussion_r94535925) [](ancestors = 94535925)